### PR TITLE
[DOCS] Rewrite `has_child` query to use new format

### PR DESCRIPTION
**Pull request by jrodewig**
Rewrites the `has_child` query to use the new query format.

This is part of #40977, an effort to standardize documentation for query types.

## Before
<details>
 <summary>Before image</summary>
Has Child Query - Before

</details>

## After
<details>
 <summary>After image</summary>
Has Child Query - After
</details>

### CODE REVIEW

File: docs/reference/query-dsl/has-child-query.asciidoc [hunk 1]
```diff
@@ -1,135 +1,158 @@
...
+the same index using a <<parent-join,join>> field mapping.
+
+[WARNING]
+====
+Because it performs a join, the `has_child` is slow compared to other queries.
+Its performance degrades as the number of matching child documents pointing to
+unique parent documents increases. Each `has_child` query in a search can
+increase query time significantly.
+
+If you care about query performance, do not use this query. If need to use the
```
Comment (cbuescher) [inline docs/reference/query-dsl/has-child-query.asciidoc]: " If need" <- not a native speaker, but is there a verb missing? Or something else from the sentence?

File: docs/reference/query-dsl/has-child-query.asciidoc [hunk 2]
```diff
@@ -1,135 +1,158 @@
...
+`type`.
+--
 
-When set to `true` the `ignore_unmapped` option will ignore an unmapped `type`
-and will not match any documents for this query. This can be useful when
-querying multiple indexes which might have different mappings. When set to
-`false` (the default value) the query will throw an exception if the `type`
-is not mapped.
+`max_children`::
+(integer) Optional. Maximum number of child documents allowed for returned parent
```
Comment (cbuescher) [inline docs/reference/query-dsl/has-child-query.asciidoc]: Reading this I was wondering if "child documents" here means child documents that match the query condition. I might have missed that part, but maybe its worth clarifying.

**Review by cbuescher — APPROVED** [review]
LGTM, I left two suggestions but feel free to adress them at will.

File: docs/reference/query-dsl/has-child-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/has-child-query.asciidoc]: You didn't miss anything. Just a typo. Should read `If you need...` Fixed with 164d716.

File: docs/reference/query-dsl/has-child-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/has-child-query.asciidoc]: Clarified with 6f4668e

**Pull request merged**